tests/unit/IonTextReaderTest.js

```diff
@@ -125,6 +125,29 @@
       assert.equal(ionReader.fieldName(), "innerkey2");
     };
 
+      suite['Accept IVM like symbols and throw on IVMs except $ion_1_0'] = function() {
+          let shouldPass = ["$ion_1_0", "$ion_schema_1_0", "$ion_1", "$ion_1_a", "$ion_", "ion_1_"];
+          let shouldFail = ["$ion_2_0", "$ion_1_999", "$ion_999_0", "$ion_1_1", "$ion_1_00"];
+          for(let input in shouldPass) {
+              let ionReader = ion.makeReader(input);
+              try {
+                  while (ionReader.next() !== undefined) {}
+              } catch(error) {
+                  throw new Error("Allowable IVM like symbol threw an error " + error);
+              }
+          }
+
+          for(let input in shouldFail) {
+              let ionReader = ion.makeReader(input);
+              let caughtError = false;
+              try {
+                  while (ionReader.next() !== undefined) {}
+              } catch(error) {
+                caughtError = true;
+              }
+              if(!caughtError) throw new Error("Unsupported IVM symbol did not throw an error.");
```

Comment:
if we move this inside the try block immediately after the while loop, the `caughtError` variable can be removed

Reply:
nice